### PR TITLE
Port several additional tests

### PR DESCRIPTION
**Pull request by sharwell**
* Port `TestVocabulary`
* Port `TestXPath`
    * Fix a small bug it revealed
    * Update documentation in some of the XPath types
* Remove tests that won't be ported
* Remove duplicate tests

### CODE REVIEW

File: src/tree/Trees.ts [hunk 1]
```diff
@@ -110,9 +121,11 @@ export class Trees {
 	 *
 	 *  @since 4.5.1
 	 */
+	public static getAncestors(t: ParseTree): ParseTree[];
+	public static getAncestors(t: Tree): Tree[];
```
Comment (BurtHarris) [inline src/tree/Trees.ts]: Nice

File: src/tree/xpath/XPath.ts [hunk 2]
```diff
@@ -27,44 +28,38 @@ import { XPathWildcardElement } from "./XPathWildcardElement";
...
+ * ```
+ * let p = new XPath(parser, pathString);
+ * return p.evaluate(tree);
+ * ```
  *
- * <p>
- * See {@code org.antlr.v4.test.TestXPath} for descriptions. In short, this
- * allows operators:</p>
+ * See `TestXPath` for descriptions. In short, this
+ * allows operators:
```
Comment (BurtHarris) [inline src/tree/xpath/XPath.ts]: Is there web-based documentation that explains the API.   This isn't covered in the latest ANTLR book, right?

File: src/tree/xpath/XPath.ts [hunk 3]
```diff
@@ -147,8 +142,8 @@ export class XPath {
 	}
 
 	/**
-	 * Convert word like {@code *} or {@code ID} or {@code expr} to a path
-	 * element. {@code anywhere} is {@code true} if {@code //} precedes the
+	 * Convert word like `*` or `ID` or `expr` to a path
```
Comment (BurtHarris) [inline src/tree/xpath/XPath.ts]: I'm pretty ignorant about JSDoc, but I remember reading in the docs that JSDoc has a Markdown parser as an add-in, but that by default, it [only looks for markdown in specific JSDoc tags.](http://usejsdoc.org/plugins-markdown.html).   

I like the idea of markdown better than tags like @code, but I don't see that this is in any of the named tags.    I don't know if this is a problem, do you know more than I, or someone else who might?

Comment (sharwell) [reply src/tree/xpath/XPath.ts]: I just just going based on what Quick Info (hover tips) looks like in VS Code. It's hard to tell what's right and what's wrong.

File: src/tree/xpath/XPath.ts (hunk 2 above)
Comment (sharwell) [reply src/tree/xpath/XPath.ts]: I'm not sure for this. I never really used the feature much.

File: src/tree/xpath/XPath.ts [hunk 4]
```diff
@@ -27,44 +28,38 @@ import { XPathWildcardElement } from "./XPathWildcardElement";
...
+ * See `TestXPath` for descriptions. In short, this
+ * allows operators:
  *
- * <dl>
- * <dt>/</dt> <dd>root</dd>
- * <dt>//</dt> <dd>anywhere</dd>
- * <dt>!</dt> <dd>invert; this must appear directly after root or anywhere
- * operator</dd>
- * </dl>
+ * | | |
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/tree/xpath/XPath.ts]: I don't quite understand the question, but this is like the other cases where I was making something work for hover tips in VS Code.

**Pull request merged**